Add LaunchPad connector

bwitherspoon opened the pull request:
The LaunchPad / BoosterPack connector is found on TI LaunchPad development kits and BoosterPack expansion modules. This PR adds support for the LaunchPad connector which allows shields (e.g. BoosterPacks) to use aliases for the connector interfaces and mappings for the GPIOs. This is similar to what is being done for the Arduino connectors.

dbkinder reviewed (changes requested):
Should there be documentation as well?

bwitherspoon commented:
> Should there be documentation as well?

I haven't included any documentation in this PR. Do I need to add some?

I originally had this based on another PR which has now been merged and this PR rebased. Because of this several CODEOWNERS got pulled in due to those commits.

dbkinder reviewed (approved):
if no doc changes are needed, I'll get out of the way...

galak commented on dts/bindings/gpio/launchpad-header.yaml:
```diff
@@ -0,0 +1,25 @@
+      type: string
+      category: required
+      description: compatible strings
+      constraint: "launchpad-header"
```
probably best to call this "ti,launchpad-header".

galak commented on dts/bindings/gpio/launchpad-header.yaml:
```diff
@@ -0,0 +1,25 @@
+      category: required
+      description: compatible strings
+      constraint: "launchpad-header"
+      generation: define
```
remove "generation:"

galak commented on dts/bindings/gpio/launchpad-header.yaml:
```diff
@@ -0,0 +1,25 @@
+#
+---
+title: LaunchPad GPIO header
+version: 0.1
```
remove version

galak commented on dts/bindings/gpio/launchpad-header.yaml:
```diff
@@ -0,0 +1,25 @@
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+---
```
remove `---`

galak reviewed (changes requested):
Looks generally fine, a few cleanups.  I assume there are some "shields" you plan on adding?

vanti commented on dts/bindings/gpio/launchpad-header.yaml:
```diff
@@ -0,0 +1,25 @@
+    gpio-map:
+      type: compound
+      category: required
+
```
I am a newbie when it comes to dts files. But looking at the arduino ones, should "#gpio-cells" be listed as a property here? The property is being used on line 10 of the dtsi files.

vanti replied:
LGTM otherwise

galak replied:
I introduced `gpio-nexus.yaml` so including that like `arduino-header-r3.yaml` does will cover it.

galak commented:
@bwitherspoon plans to update this?

bwitherspoon commented:
I had plans to add some shields, but this work has stalled. I am going to close this PR and reopen when I have time to complete the shields.

bwitherspoon commented:
I have started submitting PRs for the various shields I have worked with. This GPIO mapping is essential for configuring the interrupt lines on the various boards.

I have updated this connector to imitate how the arduino connector is now done and in the process addressed the mentioned issues.

bwitherspoon commented:
Also are there any opinions on if this should be called BoosterPack connector instead of LaunchPad connector?

bwitherspoon committed:
dts: bindings: add GPIO binding for BoosterPack header

This header is found on TI LaunchPad development kits and BoosterPack
expansion modules. This binding allows boards to define mappings from
header pins to device GPIOs.

Signed-off-by: Brett Witherspoon <spoonb@cdspooner.com>

bwitherspoon committed:
boards: cc1352r1_launchxl / cc26x2r1_launchxl: add BoosterPack connector

Add BoosterPack connector information to the DTS of the CC1352 and
CC26x2 LaunchPad boards. This allows shields (e.g. BoosterPacks) to use
aliases for the connector interfaces and mappings for the GPIOs. This is
similar to what is being done for the Arduino connectors.

Signed-off-by: Brett Witherspoon <spoonb@cdspooner.com>

vanti commented:
> Also are there any opinions on if this should be called BoosterPack connector instead of LaunchPad connector?

I think I have a preference for BoosterPack connector given it is referred to as "BoosterPack plug-in module connectors" on the [page for the TI LaunchPad](http://www.ti.com/tool/LAUNCHXL-CC1352R1). I also see some existing documentation (e.g. boards/arm/cc26x2r1_launchxl/doc/index.rst) where it is referred to as BoosterPack connector, so it is good to be consistent.

erwango commented on dts/bindings/gpio/ti,launchpad-header.yaml:
```diff
@@ -0,0 +1,11 @@
+
+description: |
+    This is a representation of GPIO pin nodes exposed as headers on TI LaunchPads
+
```
Can you provide more documentation  (cf recent https://github.com/zephyrproject-rtos/zephyr/pull/20555) ?
(a link to TI website would work).
Ideally one day bindings description could be exported to doc.zephyr....

bwitherspoon replied:
I added some documentation modeled after #20555.

bwitherspoon commented:
> > Also are there any opinions on if this should be called BoosterPack connector instead of LaunchPad connector?
> 
> I think I have a preference for BoosterPack connector given it is referred to as "BoosterPack plug-in module connectors" on the [page for the TI LaunchPad](http://www.ti.com/tool/LAUNCHXL-CC1352R1). I also see some existing documentation (e.g. boards/arm/cc26x2r1_launchxl/doc/index.rst) where it is referred to as BoosterPack connector, so it is good to be consistent.

Agreed. I renamed this boosterpack.

vanti commented on dts/bindings/gpio/ti,boosterpack-header.yaml:
```diff
@@ -0,0 +1,34 @@
+    through 40. The BoosterPack pinout is depicted below:
+
+    1  3.3V     21 5V                    40 GPIO 20 GND
+    2  Analog   22 GND                   39 GPIO 19 GPIO / SPI CS
```
Pin 18 should be "GPIO/SPI CS", not pin 19.

bwitherspoon replied:
One off. Fixed.

bwitherspoon commented:
This change has already been merged indirectly.
